### DISCUSSION TITLE
Web Push Subscriptions returns 422

### DISCUSSION THREAD

**Discussion by manuhazen** [magicbell/community#135] [Questions] [answered]
Hello, I have an issue implementing web push notifications.

I have implemented all the steps and the service worker has been properly installed in the app. However, when requesting permission for push notifications, we encounter a 422 error.

For the API, there's no mention of the endpoint the button call when you press it. [Documentation](https://www.magicbell.com/blog/web-push-notifications-from-your-domain).

![image](https://github.com/magicbell-io/public/assets/10765746/659d7856-7a8f-485d-a041-eca3d7e309c5)

![image](https://github.com/magicbell-io/public/assets/10765746/f7de905f-d6e7-4715-b263-20cbabe7a1f9)

Return

```javascript
{"errors":[{"message":"must exist"}]}
```

**Comment by unamashana** [comment, marked as answer]
Sorry about the lack of proper documentation. This is one of our newer packages and we are running behind on a better README. From the screenshot I can see that you are using the user auth headers, whereas this endpoint requires a JWT token (it's an oddball). You can get that by fetching `GET /config` with user headers. Once you have it, you can use the token as illustrated on https://github.com/magicbell-io/webpush-test/blob/main/src/components/subscriber.tsx#L23.

Happy to answer any questions.
